### DISCUSSION TITLE
Programmatically switching theme doesn't work

### DISCUSSION THREAD

**Discussion by zadam** [excalidraw/excalidraw#5212] [Q&A] [answered]
Hi, I'm trying to set the theme programmatically, like this:

```javascript
               React.createElement(Excalidraw.default, {
                    ...
                    theme: "light", // it doesn't matter if I use THEME.LIGHT or literal value
                    ...
                })
```

But the dark theme remains. Interestingly, button to change the theme manually disappears:

![image](https://user-images.githubusercontent.com/617641/169410266-f874758e-4c8b-4c5b-85e4-1b8bce22de54.png)

If I comment out the `theme: "light"` line, the button is back:

![image](https://user-images.githubusercontent.com/617641/169410391-b2f7ce4c-52f2-4689-9cfb-8bcd14e11722.png)

**Comment by dwelle** [comment, marked as answer]
Hi. Yes, the `theme` component prop allows the host app to control the theme, and as such disables the UI button.

If you want to set the theme without making it "controlled", you can use the [`updateScene()`](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#updatescene) API:

```
excalidrawAPI.updateScene({ appState: { theme: "light" } })
```

To set the theme on initial load, set the `appState` in [initialdata](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#initialdata).

**Reply by zadam** [reply]
Hi, thanks for response. To be clear, my problem isn't that the button disappears, this was for me only a sign that the setting was not *completely* ignored.

My problem is that even after setting `theme: "light"`, the theme stays dark.

I also tried `excalidrawAPI.updateScene({ appState: { theme: "light" } })` and setting the `appState` in `initialData`, but whatever I try, the excalidraw just stays dark. The only way to switch the theme is manually with the button (when it's not "controlled").

**Reply by zadam** [reply]
Nevermind, found my issue. I forgot I'm persisting and then loading the whole appState, which includes `theme` property which overwrote everything I've set.
